test/e2e/functional/tests/e2e.go

```diff
@@ -163,6 +163,9 @@ var _ = Describe("metallb", func() {
 
 			Expect(testclient.Client.Create(context.Background(), addresspool)).Should(Succeed())
 
+			defer (func() {
+				_ = testclient.Client.Delete(context.Background(), addresspool)
```

Comment:
I think we should check the returned error, if it's different than not found we should fail. Maybe also call for assertConfigMapMatchYAML? 
If this test failed to delete the addresspool or if the next test started before it was actually deleted, the next test might be creating an addresspool with the same IP range and it will be blocked by the addresspool validation webhook. 
Same for the other places in the code that are doing this addresspool cleanup.

Reply:
In addition, I see in some places we just delete the addresspool and checking the returned error, while in others we also wait for it to be deleted.